lib/middleware/request-handler.ts

```diff
@@ -163,66 +163,76 @@ function onSubscribe(req: Interface.IOurRequest,
                      res: Interface.IOurResponse,
                      next: restify.Next): void
 {
+    // Validate input options
```

Comment:
Generally, it may be a better approach to do the validation in one past and map down the valid subscriptions to subscribe in another pass.  It separates the steps clearly for subscriptions instead of conflating them in one massive _.forEach that may return early and has unclear side-effects.